
From: Andreas Henriksson <andreas@fjortis.info>

When Antonino A.  Daplas posted his "fbdev: Initialize i810fb after
agpgart" patch he said that the ugly agp initialization hack for intel agp
shouldn't be needed but that he couldn't test it.

I have tested the framebuffer updates and additionally removed the
initialization hack and it does indeed work.

Signed-off-by: Andreas Henriksson <andreas@fjortis.info>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/char/agp/intel-agp.c |   10 +---------
 1 files changed, 1 insertion(+), 9 deletions(-)

diff -puN drivers/char/agp/intel-agp.c~fbdev-remove-i810fb-explicit-agp-initialization-hack drivers/char/agp/intel-agp.c
--- 25/drivers/char/agp/intel-agp.c~fbdev-remove-i810fb-explicit-agp-initialization-hack	Wed Sep 15 14:59:52 2004
+++ 25-akpm/drivers/char/agp/intel-agp.c	Wed Sep 15 14:59:52 2004
@@ -1781,16 +1781,8 @@ static struct pci_driver agp_intel_pci_d
 	.resume		= agp_intel_resume,
 };
 
-/* intel_agp_init() must not be declared static for explicit
-   early initialization to work (ie i810fb) */
-int __init agp_intel_init(void)
+static int __init agp_intel_init(void)
 {
-	static int agp_initialised=0;
-
-	if (agp_initialised == 1)
-		return 0;
-	agp_initialised=1;
-
 	return pci_module_init(&agp_intel_pci_driver);
 }
 
_
